### PR TITLE
chore: harden v2 release checks

### PR DESCRIPTION
**Pull request by jdx**
## Summary

- test all features in CI and enforce warning-free clippy across all targets
- package and verify the crate in CI
- test the declared Rust 1.85 MSRV
- include the license, README, changelog, examples, tests, and negative compatibility corpus in the crate tarball
- update README dependency examples for v2
- document the exact v2 compatibility profile and remaining exclusions
- add a local `mise run release:check` task

## Why

The v2 stack changes public APIs and expands the language surface, so the release gate should cover serde, the published artifact, the supported compiler floor, and strict linting rather than only default-feature tests.

## Impact

The crate now declares Rust 1.85 as its minimum supported version. Runtime behavior is unchanged; this PR tightens CI, packaging, and release documentation.

## Stack

- built on #91
- merge the compatibility-correctness PR first

## Checks

- `cargo test --all-features` (115 unit tests, 2 compatibility tests, 10 doctests)
- `cargo +1.85.0 test --all-features`
- `cargo clippy --all-targets --all-features -- -D warnings`
- `cargo package --allow-dirty` (39 files, package verification passed)
- confirmed `LICENSE`, docs, examples, tests, and both compatibility corpora are packaged
- `git diff --check`

_This PR description was generated by OpenAI Codex._

---

> [!NOTE]
> **Low Risk**
> Mostly CI, packaging, and documentation; runtime behavior is unchanged aside from trivial lint-only code edits.
> 
> **Overview**
> Tightens the **v2 release gate** so CI and local tasks exercise the full feature set, enforce warning-free Clippy, verify the published crate, and run tests on the declared compiler floor.
> 
> **CI** now runs `cargo build`/`test` with `--all-features`, `cargo package`, and `cargo +1.85.0 test --all-features` after installing Rust 1.85. **`Cargo.toml`** sets `rust-version = "1.85"` and expands the `include` list so LICENSE, README, changelog, examples, tests, and compatibility corpora ship in the tarball.
> 
> **Developer workflow**: `mise` lint uses `clippy --all-targets --all-features -D warnings`; default tests use `--all-features`; new **`mise run release:check`** runs tests plus `cargo package`.
> 
> **Docs**: README documents the v2 compatibility profile (supported types vs known gaps) and updates serde examples to **`expr-lang` v2**.
> 
> Minor **Clippy-driven** source tweaks only (e.g. `ctx.get(id)`, `args.iter()`, trace formatting, test literals `3.125` instead of `3.14`); no intended evaluation semantics change.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit eb1e49cdadd72fd358d4f0a13b596deef76b5e05. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,7 +55,7 @@ fn main() {
 
 ```toml
 [dependencies]
-expr-lang = { version = "1", features = ["serde"] }
+expr-lang = { version = "2", features = ["serde"] }
```
[the remark this answers is not in the document]
Comment (jdx) [reply README.md:58]: Keeping the v2 dependency examples intentionally: this PR is the release-hardening top of the v2 stack, and v2 is the next planned release. Reverting the examples to v1 would leave the README stale when this stack is released. _This comment was generated by OpenAI Codex._
